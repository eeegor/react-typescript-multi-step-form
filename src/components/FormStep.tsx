/**
 * Component FormStep
 */

import * as React from 'react';
import './FormStep.scss';
import { Form } from './Form';

export interface Props {
	className?: string;
	children?: React.ReactNode;
	id: string;
	onSubmit: (form: React.FormEvent<HTMLFormElement>) => void;
}

export interface State {}

export class FormStep extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);
	}

	render(): JSX.Element {
		const { id, children, className, onSubmit } = this.props;

		return (
			<Form id={id} className="form-step" onSubmit={form => onSubmit && onSubmit(form)}>
				{children}
			</Form>
		);
	}
}
