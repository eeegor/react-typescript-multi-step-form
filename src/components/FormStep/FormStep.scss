.form-step {
	padding: 2rem 0;
	width: 100%;
	margin-bottom: auto;
	padding-bottom: 6rem;
}
