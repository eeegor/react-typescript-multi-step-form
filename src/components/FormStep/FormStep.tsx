/**
 * Component FormStep
 */

import React from 'react';
import classnames from 'classnames';
import { Form } from '..';
import './FormStep.scss';

export interface Props {
	className?: string;
	children?: React.ReactNode;
	id: string;
	onSubmit: (form: React.FormEvent<HTMLFormElement>) => void;
}

export const FormStep = (props: Props): JSX.Element => {
	const { id, children, className, onSubmit } = props;

	return (
		<Form
			id={id}
			className={classnames('form-step', className)}
			onSubmit={form => onSubmit && onSubmit(form)}
		>
			{children}
		</Form>
	);
};
