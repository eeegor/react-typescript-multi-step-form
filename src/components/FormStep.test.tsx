import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { FormStep } from './FormStep';

describe('FormStep', () => {
	const callback = jest.fn();

	it('renders markup correctly', () => {
		const component = <FormStep id="first" onSubmit={callback} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('submit event works', () => {
		const wrapper = mount(
			<FormStep id="first" onSubmit={callback}>
				Hi
			</FormStep>
		);
		wrapper.find('form').simulate('submit');
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
