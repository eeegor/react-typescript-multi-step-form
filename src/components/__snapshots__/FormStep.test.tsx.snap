// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStep renders markup correctly 1`] = `
<form
  className="form form-step"
  id="first"
  onSubmit={[Function]}
>
  <div
    className="form__container"
  />
</form>
`;
